controllers/clustergroupupgrade_controller.go

```diff
@@ -224,20 +224,24 @@ func (r *ClusterGroupUpgradeReconciler) Reconcile(ctx context.Context, req ctrl.
 					Message: "The ClusterGroupUpgrade CR has all upgrade policies compliant",
 				})
 			}
+		} else if readyCondition.Reason == "UpgradeTimedOut" {
+
```

Comment:
Was wondering if we could add a Log here, just to get an idea of what's happening. Your choice if it makes sense.